src/libraries/System.Text.Json/Common/ReflectionExtensions.cs

```diff
@@ -228,7 +228,14 @@ private static string GetBaseNameFromGenericTypeDef(Type genericTypeDef)
         public static bool IsVirtual(this PropertyInfo? propertyInfo)
         {
             Debug.Assert(propertyInfo != null);
-            return propertyInfo != null && (propertyInfo.GetMethod?.IsVirtual == true || propertyInfo.SetMethod?.IsVirtual == true);
+            return propertyInfo != null &&
+                (propertyInfo.GetMethod?.IsVirtual == true || propertyInfo.SetMethod?.IsVirtual == true
+#if BUILDING_SOURCE_GENERATOR
+                // The Roslyn PropertyInfo.IsVirtual extension is based on ISymbol and does not match
+                // reflection semantics which includes IsAbstract, so add that here.
+                || propertyInfo.GetMethod?.IsAbstract == true || propertyInfo.SetMethod?.IsAbstract == true
+#endif
```

Comment:
Would an alternative here be to add `attributes |= MethodAttributes.Virtual;` on L41 here?

https://github.com/dotnet/runtime/blob/a8eaaa7473d462fdccccf38504bfc7ad803347b8/src/libraries/System.Text.Json/gen/Reflection/RoslynExtensions.cs#L40

Reply:
Thanks. For some reason, I thought those extensions were actually in Roslyn code, not STJ.